### DISCUSSION TITLE
Combining LocalView and RemoteView - issue after update

### DISCUSSION THREAD

**Discussion by juanprietob** [Kitware/trame#487] [Q&A] [answered]
A reference to the previous [issue](https://github.com/Kitware/trame/discussions/485)

The video describes the steps to reproduce the bug/issue. 

1. The app displays an Ultrasound video on the left side and a series of 3D renderers. 
2. I want to dynamically add components to the 3D renderer. 
3. When I add a component (plane) to the renderer, the image view stops working. 
4. After a resize event of the window or similar event, the image viewer starts working again. 

The ImageViewer is running as a VtkRemoteView and the Renderer is a vtkLocalView (see previous issue). 

What is the correct way to update a Local or a Remote view?

https://github.com/Kitware/trame/assets/7086191/95ce1b2d-fe7e-4e79-91ee-c11e81163564

**Comment by jourdain** [comment, marked as answer]
3. Which view local or remote?

The correct way to update any view (local or remote) is to call the `html_view.update()` method on them. Keep in mind you can use the controller to centralize those calls more easily.

```python
with vtk_widgets.VtkRemoteView(rw_1) as view:
    ctrl.view_update.add(view.update)

with vtk_widgets.VtkLocalView(rw_2) as view:
    ctrl.view_update.add(view.update)

# update both views
ctrl.view_update()
```

**Reply by juanprietob** [reply]
This did it!

**Reply by juanprietob** [reply]
Thank you so much for your help

**Comment by jourdain** [comment]
ok, I finally saw what you meant in your video... ;-)
Not sure why the remote view stoped working after you added something in the scene...

Could you provide more information on what you did to add that plane? Did you rebuild the UI? If so that could explain why you got a glitch on the widget. Otherwise, it is hard to tell.
